src/main.rs

```diff
@@ -204,11 +210,34 @@ fn install(args: InstallOpts) -> Result<()> {
             extra_crates = Some(crates);
         };
     } else {
-        exports.extend(install_gcc_targets(&targets, &host_triple)?);
+        for target in &targets {
+            let gcc = Gcc::new(target, &host_triple);
+            to_install.push(Box::new(gcc));
+        }
     }
 
     if let Some(ref extra_crates) = &extra_crates {
-        install_extra_crates(extra_crates)?;
+        for krate in extra_crates {
+            to_install.push(Box::new(krate.to_owned()));
+        }
+    }
+
+    // With a list of applications to install, install them all in parallel.
+    let (tx, mut rx) = mpsc::channel::<Vec<String>>(32);
+    let installable_items = to_install.len();
+    for app in to_install {
+        let tx = tx.clone();
+        tokio::spawn(async move {
+            let res = app.install().await;
+            let res = res.unwrap();
+            tx.send(res).await.unwrap();
+        });
+    }
+
+    // Read the results of the install tasks as they complete.
+    for _ in 0..installable_items {
+        let names = rx.recv().await.unwrap();
+        exports.extend(names);
     }
```

Comment:
```suggestion
    while let Some(names) = rx.recv().await {
        exports.extend(names);
    }
```

What you have is fine of course, but maybe this is a bit cleaner?

Reply:
I tried this at first, but it was not working correctly (not really sure why). If I use the `while` loop, the installation will never end, as it gets stuck in the `while` loop.